### PR TITLE
feat(logpacker): add rosbag recoding api

### PR DESCRIPTION
**Pull request by ito-san**
## Related Links

[TIER IV INTERNAL SLACK](https://star4.slack.com/archives/CRUE57C30/p1710504448755029?thread_ts=1710494084.147749&cid=CRUE57C30)

## Description
The request command should use ROS2 service instead of topic, and the status should be exchanged in numerical format rather than in string.

Changes
- Request to start/stop rosbag recording
  - `/bagpacker/record std_msgs/Bool` has been discontinued.
  - Replace this with `/api/autoware/set/rosbag_record tier4_external_api_msgs::srv::SetRosbagRecord`
- Status of rosbag recoding
  - `/bagpacker/state std_msgs/msg/String` has been discontinued.
  - Replace this with `/api/autoware/get/rosbag_record_status tier4_external_api_msgs::msg::RosbagRecordStatus`
- Status of whether the `ros2 bag record` command is currently running or not.
   - `/bagpacker/condition std_msgs/Bool` has been discontinued since this duplicates the above.

## Pre-Review Checklist for the PR Author

**PR Author should check the checkboxes below when creating the PR.**

- [x] Code is properly formatted
- [x] Assign PR to reviewer

## Checklist for the PR Reviewer

**Reviewers should check the checkboxes below before approval.**

- [x] Commits are properly organized and messages are according to the guideline
- [ ] Code is properly formatted
- [x] PR title describes the changes

## Post-Review Checklist for the PR Author

**PR Author should check the checkboxes below before merging.**

- [x] All open points are addressed and tracked via issues or tickets
- [ ] Write [release notes][release-notes]

## CI Checks

- **Build and test for PR**: Required to pass before the merge.
- **Check spelling**: NOT required to pass before the merge. It is up to the reviewer(s). See [here][spell-check-dict] if you want to add some words to the spell check dictionary.

[release-notes]: https://tier4.atlassian.net/wiki/spaces/AIP/pages/563774416
[spell-check-dict]: https://github.com/tier4/autoware-spell-check-dict#how-to-contribute

### CODE REVIEW

**Review by 0x126 — APPROVED** [review]
LGTM

**Review by tsugi1203 — APPROVED** [review]
LGTM

**Review by tkimura4 — APPROVED** [review]
LGTM

**Review by shmpwk — APPROVED** [review]

**Review by yabuta — APPROVED** [review]
LGTM

File: tier4_external_api_msgs/msg/RosbagRecordStatus.msg [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# constants
+uint8 INITIALIZING=0
+uint8 WAITING_VEHICLE_INFO=1
+uint8 READY=2
+uint8 RECORDING=3
+uint8 ERROR=4
```
Comment (yabuta) [inline tier4_external_api_msgs/msg/RosbagRecordStatus.msg:6]: すみません、ApproveにしてしまたのですがERRORになるパターンとしては「disk容量が足りない」以外に何が有り得そうでしょうか？

**Comment by mitsudome-r** [comment]
Memo:
* I would like to have confirmation with the topic names for `/api/autoware/set/rosbag_record` and `/api/external/get/rosbag_record_status` to see if they should share the same namespace or not. (This can be discussed when we make the changes to logpacker)
* Please update the documentation for TIER IV API page https://github.com/tier4/autoware-documentation (https://tier4.github.io/autoware-documentation/tier4-main/design/autoware-interfaces/prototyping/) so that we can track the usecase of this new service and message.
* (cc. @yukkysaito ) We should consider refactoring [tier4_api_msgs](https://github.com/tier4/tier4_autoware_msgs/tree/tier4/universe/tier4_api_msgs) and [tier4_external_api_msgs](https://github.com/tier4/tier4_autoware_msgs/tree/tier4/universe/tier4_external_api_msgs) . Depending on the discussion we might have to change the package name (so the namespace of the message in the code) later.

**Review by mitsudome-r — APPROVED** [review]

File: tier4_external_api_msgs/msg/RosbagRecordStatus.msg (hunk 1 above)
Comment (ito-san) [reply tier4_external_api_msgs/msg/RosbagRecordStatus.msg:6]: @yabuta 
容量不足以外では、以下条件でERRORに遷移します。
- 設定が不正の時
- rosbag record中にディスクが接続されてない状態になった時(書き込み先のフォルダが存在しない）

Comment (yabuta) [reply tier4_external_api_msgs/msg/RosbagRecordStatus.msg:6]: なるほど。ありがとうございます

**Review by yabuta — APPROVED** [review]
LGTM

**Comment by ito-san** [comment]
@mitsudome-r 
> /api/autoware/set/rosbag_record and /api/external/get/rosbag_record_status

The namespace should be shared and it should be `/api/autoware`. I made changes to the description.

**Comment by ito-san** [comment]
@mitsudome-r

> documentation for TIER IV API page

Filed a new PR, please see https://github.com/tier4/autoware-documentation/pull/44.

**Pull request merged**